webapp/src/pages/EditUserPasswordPage/EditUserPasswordPage.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  AvatarPageTemplate,
+  ButtonVariant,
+  Input,
+  InputVariant,
+  Loading,
+} from '../../shared/components';
+import { AVATAR_EP_URL, USER_ME_URL, USER_URL } from '../../shared/urls';
+import { useParams } from 'react-router-dom';
+import { useNavigation } from '../../shared/hooks/UseNavigation';
+import React, { useState } from 'react';
+import { ResponseError, UpdateUserDto } from '../../shared/generated';
+import { usersApi } from '../../shared/services/ApiService';
+import { useNotificationContext } from '../../shared/context/NotificationContext';
+import NotFoundPage from '../NotFoundPage/NotFoundPage';
+import { useAuth } from '../../shared/hooks/UseAuth';
+
+export default function EditUserPasswordPage() {
+  const { username } = useParams();
+  const { authUser, isLoading } = useAuth(username);
+  const { navigate } = useNavigation();
+  const { warn, notify } = useNotificationContext();
+
+  const initialFormValues: UpdateUserDto = {
+    oldPassword: '',
+    password: '',
+    confirmationPassword: '',
+  };
+  const [formValues, setFormValues] =
+    useState<UpdateUserDto>(initialFormValues);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormValues((previousValues: any) => {
+      return { ...previousValues, [name]: value };
+    });
+  };
+
+  function hasValidFormValues() {
+    if (formValues.oldPassword === '') {
+      warn('Old password can not be empty');
+      return false;
+    }
+    if (formValues.password === '') {
+      warn('Password can not be empty');
+      return false;
+    }
+    if (formValues.password !== formValues.confirmationPassword) {
+      warn('Password and Confirmation Password must match');
+      return false;
+    }
+    return true;
+  }
+
+  const handleRequestError = async (error: unknown) => {
+    let errMessage = '';
+    if (error instanceof ResponseError) {
+      const responseBody = await error.response.json();
+      if (responseBody.message) {
+        errMessage = responseBody.message;
+      } else {
+        errMessage = error.response.statusText;
+      }
+    } else if (error instanceof Error) {
+      errMessage = error.message;
+    }
+    warn(errMessage);
+  };
+
+  async function updatePassword() {
+    if (!authUser || !hasValidFormValues()) {
+      return;
+    }
+    try {
+      await usersApi.userControllerUpdateCurrentUser({
+        updateUserDto: formValues,
+      });
+      notify('Password successfully updated');
+      navigate(USER_ME_URL);
+    } catch (error) {
+      handleRequestError(error);
```

Comment:
Maybe handleRequestError should be in the shared code. Not sure in which file. It's used (or could be used) everywhere in our UI.

Reply:
Also 

> Missing await for an async function call

Reply:
I wil leave a TODO, similiar to another one in the edit chat details